lib/Models/CsvCatalogItem.js

```diff
@@ -672,25 +722,40 @@ function createRegionLookupFunc(csvItem) {
 
     // set color for each code
     var row, index, codeMap = [];
-    if (!defined(csvItem.recs)) {
-        for (row = 0; row < codes.length; row++) {
-            index = ids.indexOf(codes[row]);
-            if (index >= 0 && colors[index][3] === 0) {
-                colors[index] = dataSource._mapValue2Color(vals[row]);
-            }
-        }
-    } else {
-        for (var i = 0; i < csvItem.recs.length; i++) {
-            row  = csvItem.recs[i];
+    var multimatches = {};
+    var failedmatches = {};
+    var successes = 0, totalrows=0;
+
+    var userecs = defined(csvItem.recs);
+    for (var i = 0; i < (userecs ? csvItem.recs.length : codes.length); i++) {
+        if (userecs) {
+            row = csvItem.recs[i];
             codeMap[i] = codes[row];
+        } else {
+            row = i;
+        }
+        totalrows++;
+        if (typeof codes[row] === "number") {
             index = ids.indexOf(codes[row]);
-            if (index >= 0 && colors[index][3] === 0) {
-                colors[index] = dataSource._mapValue2Color(vals[row]);
-            }
+        } else {
+            index = ids.indexOf(String(codes[row]).toLowerCase());
```

Comment:
I don't think this is new, but if performance is a concern it looks it should be easy to replace this linear search with a hash lookup.  I have to imagine that would make a big difference with e.g. SA1.